modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -156,19 +171,31 @@ public void triggered(SchedulerEngine.Event event) {
     // default visibility for testing purposes
     void run(ClusterState state) {
         for (DataStream dataStream : state.metadata().dataStreams().values()) {
+            clearErrorStoreForUnmanagedIndices(dataStream);
             if (dataStream.getLifecycle() == null) {
                 continue;
             }
 
+            String writeIndex = dataStream.getWriteIndex().getName();
             try {
                 maybeExecuteRollover(state, dataStream);
             } catch (Exception e) {
                 logger.error(() -> String.format(Locale.ROOT, "DLM failed to rollver data stream [%s]", dataStream.getName()), e);
+                DataStream latestDataStream = clusterService.state().metadata().dataStreams().get(dataStream.getName());
```

Comment:
Why do we get the data stream a second time, instead of using the `dataStream` as part of the `for` loop?

Reply:
This is trying to avoid a bit of noise - notice that the data stream is re-read from the latest cluster state (not from the state that's passed in the `run` method)

If the write index for which we **attempted** to rollover the data stream has changed by the time we get to the `catch` block we don't report the error anymore (the data stream was indeed rolled over so it'd be strange to report a rollover problem for an index that's not the write index anymore).
Note that this just reduces the window of opportunity for such a situation.